README.md

```diff
@@ -1,85 +1,341 @@
+2. If you want to use Iterable Rich Notifiation Extension, you will need to add IterableAppExtension.framework to **Linked Frameworks and Libraries** section of your **app extension** target (not app target). Please note that you will have to add the IterableAppExtension.framework bundle to **both** the app target (step 1) and app extension target (step 2) of your project. In the app target it goes in the 'Embedded Binaries' section and in app extension target it goes in the 'Linked Frameworks and Libraries' section.
 
+	![Linking](./images/app-extension-linked-framework.png)
 
-# Using the SDK
-1. Initialize the API with API key. The only required parameter is `apiKey`. You may also like to set default configuration parameters.
+3. In build settings, set `Always Embed Swift Standard Libraries` setting to 'Yes'. This is required for Objective C projects.
+	
+	![Linking](./images/build-setting.png)
+ 
+
+# Initializing the SDK
+**Note:** Sample projects are included in this repo.
+	 
+- [Swift Sample Project](./sample-apps/swift-sample-app/swift-sample-app.xcworkspace)
+- [ObjC Sample Project](./sample-apps/objc-sample-app/objc-sample-app.xcworkspace)
 
+
+1. ##### Initialize the API with API key.
+	   In your app delegate, on application launch in `application:didFinishLaunchingWithOptions:` method, initialize the Iterable SDK:
+
+	Swift:
+
+	```swift
+	let config = IterableConfig()
+	config.pushIntegrationName = "<your-iterable-push-integration-name>"
+	IterableAPI.initialize(apiKey: "<your-api-key>", launchOptions: launchOptions, config:config)
+	```
+	
+	Objective-C:
+
+	```objective-c
+	IterableConfig *config = [[IterableConfig alloc] init];
+	config.pushIntegrationName = @"<your-iterable-push-integration-name>";
+	[IterableAPI initializeWithApiKey:@"<your-api-key>" launchOptions:launchOptions config:config]
+	```
+	
+	
+	See the Iterable guide on how to setup your Iterable push integration and obtain push integration name [here](https://support.iterable.com/hc/en-us/articles/115000315806-Setting-Up-iOS-Push-Notifications).	
+	
+2. ##### Set userId or email. 
+
+	Once you know the email or userId of the user, set the value.
+
+	Swift:
+	
+	```swift
+	IterableAPI.email = "user@example.com"
 	```
-	IterableAPI.initialize(apiKey: "your-api-key", optionalParameters.....)
+
+	Objective-C:
+
+	```objective-c
+	IterableAPI.email = @"user@example.com";
 	```
-  * Ideally, you will call this from inside `application:didFinishLaunchingWithOptions:` and pass in `launchOptions`. This will let the SDK automatically track a push open for you if the application was launched from a remote Iterable push notification. 
 	
-2. Once you know the email *(Preferred)* or userId of the user, set the value.
+3. ##### Send Remote Notification Token to Iterable
+
+	* See [Apple Notification Guide](https://developer.apple.com/documentation/usernotifications) regarding how to register for remote notifiations.
```

Comment:
Should we point them to specific articles about requesting a token and requesting permissions?